core/core-bitcoin.js

```diff
@@ -138,8 +138,28 @@ CoreBitcoin.prototype.asyncPollBalance = function () {
   }, this)
 }
 
+CoreBitcoin.prototype.getLastBalances = function () {
+  return this.balances
+}
+
 CoreBitcoin.prototype.asyncGetLatestBlockInfo = function () {
-  return this.blockchainAPI.asyncGetLatestBlockInfo()
+  return asink(function *() {
+    let blockInfo = yield this.blockchainAPI.asyncGetLatestBlockInfo()
+
+    let blockInfoFound = blockInfo && blockInfo.hashbuf
+    let blockChanged = blockInfoFound && (!this.lastBlockInfo || !this.lastBlockInfo.hashbuf || this.lastBlockInfo.hashbuf.toString('hex') !== blockInfo.hashbuf.toString('hex'))
+
+    if (blockChanged) {
+      this.emit('block-info', blockInfo)
```

Comment:
This seems like poor practice to me to emit an event inside a "get" method. The only thing the "get" function should do is get the value. Perhaps the proper place for this code to be is inside the blockchainAPI code, which could be set to monitor for the latest block info. And in that case, when an update is found, an event is emitted.